PlanPluginBridge/src/main/java/com/djrapitops/pluginbridge/plan/nucleus/NucleusData.java

```diff
@@ -0,0 +1,311 @@
+ */
+public class NucleusData extends PluginData {
+    private UserStorageService userStorageService = null;
+
+    public NucleusData() {
+        super(ContainerSize.TWO_THIRDS, "Nucleus");
+        setPluginIcon(Icon.called("flask").of(Color.DEEP_ORANGE).build());
+
+        Sponge.getServiceManager().provide(UserStorageService.class).ifPresent(storageService -> userStorageService = storageService);
+    }
+
+    @Override
+    public InspectContainer getPlayerData(UUID uuid, InspectContainer inspectContainer) {
+        User user = getUser(uuid);
+
+        if (user == null) {
+            inspectContainer.addValue("Data unavailable", "Could not get user data");
+            return inspectContainer;
+        }
+
+        if (NucleusAPI.getMuteService().isPresent()) {
+            NucleusMuteService muteService = NucleusAPI.getMuteService().get();
+
+            boolean muted = muteService.isMuted(user);
+            inspectContainer.addValue(getWithIcon("Muted", Icon.called("bell-slash").of(Color.DEEP_ORANGE)), muted ? "Yes" : "No");
+
+            if (muted && muteService.getPlayerMuteInfo(user).isPresent()) {
+                MuteInfo muteInfo = muteService.getPlayerMuteInfo(user).get();
+
+                String reason = HtmlUtils.swapColorsToSpan(muteInfo.getReason());
+                String start = "Unknown";
+                String end;
+                String link = "Unknown";
+
+                if (muteInfo.getMuter().isPresent()) {
+                    User operatorUser = getUser(muteInfo.getMuter().get());
+                    if (operatorUser != null) {
+                        String operator = operatorUser.getName();
+                        link = Html.LINK.parse(PlanAPI.getInstance().getPlayerInspectPageLink(operator), operator);
+                    }
+                }
+
+                if (muteInfo.getCreationInstant().isPresent()) {
+                    start = FormatUtils.formatTimeStampYear(muteInfo.getCreationInstant().get().toEpochMilli());
+                }
+
+                if (muteInfo.getRemainingTime().isPresent()) {
+                    end = FormatUtils.formatTimeStampYear(muteInfo.getRemainingTime().get().plusMillis(System.currentTimeMillis()).toMillis());
+                } else {
+                    end = "Permanent mute";
+                }
+
+                inspectContainer.addValue("&nbsp;" + getWithIcon("Operator", Icon.called("user").of(Color.DEEP_ORANGE)), link);
+                inspectContainer.addValue("&nbsp;" + getWithIcon("Date", Icon.called("calendar").of(Color.DEEP_ORANGE).of(Family.REGULAR)), start);
+                inspectContainer.addValue("&nbsp;" + getWithIcon("Ends", Icon.called("calendar-check").of(Color.DEEP_ORANGE).of(Family.REGULAR)), end);
+                inspectContainer.addValue("&nbsp;" + getWithIcon("Reason", Icon.called("comment").of(Color.DEEP_ORANGE).of(Family.REGULAR)), reason);
+            }
+        }
+
+        if (NucleusAPI.getJailService().isPresent()) {
```

Comment:
This if block could be refactored into it's own method
`ifPresent(service -> newMethod(service, inspectContainer))`